tests/Base/Producer/RecordValidatorTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+declare(strict_types=1);
+
+namespace KafkaTest\Base\Producer;
+
+use Kafka\Exception\InvalidRecordInSet;
+use Kafka\Producer\RecordValidator;
+use PHPUnit\Framework\TestCase;
+
+final class RecordValidatorTest extends TestCase
+{
+    /** @var RecordValidator */
+    private $recordValidator;
+
+    public function setUp(): void
+    {
+        $this->recordValidator = new RecordValidator();
+
+        parent::setUp();
+    }
+
+    /**
+     * @param mixed[] $record
+     * @dataProvider validateDataProvider
+     */
+    public function testValidate(?string $expectedExceptionMessage, array $record): void
+    {
+        if ($expectedExceptionMessage === null) {
+            $this->assertTrue(true);
```

Comment:
This a no go, let's move the valid scenario to another test case with the `@doesNotPerformAssertions` annotation.

Reply:
didnt know this one, thanks!